tests/framework/microvm.py

```diff
@@ -371,12 +399,24 @@ def spawn(self):
         # and leave 0.2 delay between them.
         if 'no-api' not in self._jailer.extra_args:
             self._wait_create()
+        if create_logger:
+            self._wait_for_message("Running Firecracker", 0.5)
 
     @retry(delay=0.2, tries=5)
     def _wait_create(self):
         """Wait until the API socket and chroot folder are available."""
         os.stat(self._jailer.api_socket_path())
 
+    def _wait_for_message(self, message, timeout):
```

Comment:
We could make use of the retry decorator here too. The retry decorator will re-run the function 3 times with a delay of 0.5 seconds upon catching any exception (in our case the exception will be AssertionError):
```
@retry(tries=3, delay=0.5)
def check_log_message:
'''Retry on AssertionError, raise error after 3 attempts, sleep 0.5 seconds between attempts.'''
assert message in self._log_data
```
This is just an example, you can tweak the parameters. WDYT?

Reply:
I was thinking that maybe we will use that function with a different timeout at some point, but until that, this simplifies things a lot :ok_hand:. I've updated with your suggestion.